dhis-2/dhis-support/dhis-support-system/src/main/java/org/hisp/dhis/system/util/ReflectionUtils.java

```diff
@@ -527,4 +532,49 @@ public static boolean isTranslationProperty( Property property )
     {
         return "translations".equals( property.getName() ) || "translations".equals( property.getCollectionName() );
     }
+
+    public static List<Field> findFields( Class<?> klass, Predicate<Field> predicate )
+    {
+        return getAllFields( klass ).stream()
+            .filter( predicate )
+            .collect( Collectors.toList() );
+    }
+
+    public static List<Method> findMethods( Class<?> klass, Predicate<Method> predicate )
+    {
+        return getMethods( klass ).stream()
+            .filter( predicate )
+            .collect( Collectors.toList() );
+    }
+
+    public static String getFieldName( Method method )
+    {
+        String name;
```

Comment:
Also just generally can we stop pascal style method variables? 
This always needs tracking the variable throughout the method to see if and where it is set and how and by which condition it will exit the method. 
This style also has the tendency to have you implement it unnecessary looping when a result has already been found.
It becomes much more clear having a loop that has a condition with a immediate `return` and a default case at the end having another return with a "hard coded" value like here we could write `return uncapitalize(method.getName())`.
Here I'd even think two if blocks would be much more clear, like
```java
if (name.startsWith("is")) return ...
if (name.startsWith("has") || name.startsWith("get")) return ...
return ...
``` 
Reading it like that you immediately see when we are exiting with what value.

Reply:
Yeah, this one was just moved from another place without refactor, cleaned it up a bit now